osu.Game/Screens/Select/BeatmapClearScoresDialog.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Overlays.Dialog;
+using osu.Game.Scoring;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace osu.Game.Screens.Select
+{
+    public class BeatmapClearScoresDialog : PopupDialog
+    {
+        private ScoreManager manager;
+
+        [BackgroundDependencyLoader]
+        private void load(ScoreManager beatmapManager)
+        {
+            manager = beatmapManager;
+        }
+
+        public BeatmapClearScoresDialog(BeatmapSetInfo beatmap, IEnumerable<ScoreInfo> scores, Action refresh)
```

Comment:
why do you pass in the score rather than just getting them here?

Reply:
Isn't passing them through rather than getting them here again more performant?

Reply:
in a scenario like this performance is not even a consideration

Reply:
Resolved in https://github.com/ppy/osu/pull/4000/commits/097062caf78bd7985fa4d1af9b675743191a6671